src/Common/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs

```diff
@@ -90,6 +124,39 @@ private unsafe bool TryEncryptOrDecrypt(ReadOnlySpan<byte> data, Span<byte> dest
 
             using (SafeNCryptKeyHandle keyHandle = GetDuplicatedKeyHandle())
             {
+                if (encrypt && data.Length == 0)
+                {
+                    int bufSize = RsaPaddingProcessor.BytesRequiredForBitCount(KeySize);
+                    byte[] rented = ArrayPool<byte>.Shared.Rent(bufSize);
+                    Span<byte> paddedMessage = new Span<byte>(rented, 0, bufSize);
+
+                    try
+                    {
+                        if (padding == RSAEncryptionPadding.Pkcs1)
+                        {
+                            RsaPaddingProcessor.PadPkcs1Encryption(data, paddedMessage);
+                        }
+                        else if (padding.Mode == RSAEncryptionPaddingMode.Oaep)
+                        {
+                            RsaPaddingProcessor processor =
+                                RsaPaddingProcessor.OpenProcessor(padding.OaepHashAlgorithm);
+
+                            processor.PadOaep(data, paddedMessage);
+                        }
+                        else
+                        {
+                            throw new CryptographicException(SR.Cryptography_UnsupportedPaddingMode);
+                        }
+
+                        return TryEncryptOrDecrypt(keyHandle, paddedMessage, destination, AsymmetricPaddingMode.NCRYPT_NO_PADDING_FLAG, null, encrypt, out bytesWritten);
+                    }
+                    finally
+                    {
+                        CryptographicOperations.ZeroMemory(paddedMessage);
```

Comment:
We'd need to be careful with it, since we need to make sure it's disposable (for using-goodness) and maybe Finalizable (to avoid memory leaks).

ArrayPooled, eventually, tends to be Gen2, so it doesn't move.  But if we're doing guaranteed pin we need to be careful.